ddtrace/opentelemetry/metric/meter_provider.go

```diff
@@ -0,0 +1,188 @@
+// - OTEL_METRICS_EXPORTER=none
+//
+// When disabled, returns a no-op MeterProvider that doesn't export metrics.
+//
+// Users can override these defaults by passing additional options.
+func NewMeterProvider(opts ...Option) (otelmetric.MeterProvider, error) {
+	return NewMeterProviderWithContext(context.Background(), opts...)
+}
+
+// NewMeterProviderWithContext creates a new MeterProvider with a custom context.
+func NewMeterProviderWithContext(ctx context.Context, opts ...Option) (otelmetric.MeterProvider, error) {
+	// Check if metrics are enabled via environment variables
+	if !isMetricsEnabled() {
+		// Return a no-op MeterProvider that doesn't export metrics
+		return noop.NewMeterProvider(), nil
+	}
+
+	// Apply configuration options
+	cfg := newConfig()
+	for _, opt := range opts {
+		opt.apply(cfg)
+	}
+
+	// Build Datadog-specific resource
+	res, err := buildDatadogResource(ctx, cfg.resourceOptions...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create OTLP exporter with DD defaults (supports both HTTP and gRPC)
+	exporter, err := newDatadogOTLPExporter(ctx, cfg.httpExporterOptions, cfg.grpcExporterOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build metric reader with DD defaults
+	// Note: Temporality is configured via the exporter's TemporalitySelector option
+	// The default OTLP exporter uses cumulative, but we configure delta via exporter options
+	reader := metric.NewPeriodicReader(
+		exporter,
+		metric.WithInterval(cfg.exportInterval),
+		metric.WithTimeout(cfg.exportTimeout),
+	)
+
+	// Create the MeterProvider
+	return metric.NewMeterProvider(
+		metric.WithResource(res),
+		metric.WithReader(reader),
+	), nil
+}
+
+// isMetricsEnabled checks environment variables to determine if metrics should be enabled.
+// Metrics are disabled by default and can be enabled by:
+// - Setting DD_METRICS_OTEL_ENABLED=true
+//
+// Returns false (disabled) if:
+// - DD_METRICS_OTEL_ENABLED is "false" or unset (default)
+// - OTEL_METRICS_EXPORTER is set to "none"
+func isMetricsEnabled() bool {
+	// Check OTEL_METRICS_EXPORTER first - if set to "none", always disable
+	if exporter := env.Get(envOtelMetricsExporter); exporter != "" {
+		exporter = strings.ToLower(strings.TrimSpace(exporter))
+		if exporter == "none" {
+			return false
+		}
+	}
+
+	// Check DD_METRICS_OTEL_ENABLED (default: false/disabled)
+	metricsEnabled := env.Get(envDDMetricsOtelEnabled)
+	if metricsEnabled == "" {
+		// If not set, default to disabled
+		return false
+	}
+
+	// If explicitly set, respect the value
+	metricsEnabled = strings.ToLower(strings.TrimSpace(metricsEnabled))
+	if metricsEnabled == "false" || metricsEnabled == "0" {
+		return false
+	}
+	if metricsEnabled == "true" || metricsEnabled == "1" {
+		return true
+	}
+
+	// Invalid value, default to disabled
+	return false
+}
+
+// IsNoop returns true if the given MeterProvider is a no-op provider that doesn't export metrics.
+func IsNoop(mp otelmetric.MeterProvider) bool {
```

Comment:
We don't need to export this method, in fact we shouldn't.
```suggestion
func isNoop(mp otelmetric.MeterProvider) bool {
```